tests/utilities/test_app.py

```diff
@@ -177,3 +217,103 @@ def disable_input(self) -> None:
+    TIME_RESOLUTION = 10_000_000
+
+    def __init__(
+        self,
+        *,
+        ticks_granularity_fps: int = 60,
+    ):
+        self._ticks_granularity_fps = ticks_granularity_fps
+        self._single_tick_duration = int(self.TIME_RESOLUTION / ticks_granularity_fps)
+        self._start_time: int = -1
+        self._current_time: int = -1
+        self._pending_sleep_events: list[tuple[int, asyncio.Event]] = []
+
+    def get_time_no_wait(self) -> float:
+        if self._current_time == -1:
+            self._start_clock()
+
+        return self._current_time / self.TIME_RESOLUTION
+
+    async def sleep(self, seconds: float) -> None:
+        event = asyncio.Event()
+        internal_waiting_duration = int(seconds * self.TIME_RESOLUTION)
+        target_event_monotonic_time = self._current_time + internal_waiting_duration
+        self._pending_sleep_events.append((target_event_monotonic_time, event))
+        # Ok, let's wait for this Event
+        # (which can only be "unlocked" by calls to `advance_clock()`)
+        await event.wait()
+
+    async def advance_clock(self, seconds: float) -> None:
+        """
+        Artificially advance the Textual clock forward.
+
+        Args:
+            seconds: for each second we will artificially tick `ticks_granularity_fps` times
+        """
+        if self._current_time == -1:
+            self._start_clock()
+
+        ticks_count = ceil(seconds * self._ticks_granularity_fps)
+        activated_timers_count_total = 0  # useful when debugging this code :-)
+        for tick_counter in range(ticks_count):
+            self._current_time += self._single_tick_duration
+            activated_timers_count = self._check_sleep_timers_to_activate()
+            activated_timers_count_total += activated_timers_count
+            # Now that we likely unlocked some occurrences of `await sleep(duration)`,
+            # let's give an opportunity to asyncio-related stuff to happen:
+            if activated_timers_count:
+                await let_asyncio_process_some_events()
+
+        await let_asyncio_process_some_events()
+
+    def _start_clock(self) -> None:
+        # N.B. `start_time` is not actually used, but it is useful to have when we set breakpoints there :-)
+        self._start_time = self._current_time = int(monotonic() * self.TIME_RESOLUTION)
+
+    def _check_sleep_timers_to_activate(self) -> int:
+        activated_timers_count = 0
+        for i, (target_event_monotonic_time, event) in enumerate(
+            self._pending_sleep_events
+        ):
+            if self._current_time < target_event_monotonic_time:
+                continue  # not time for you yet, dear awaiter...
+            # Right, let's release this waiting event!
+            event.set()
+            activated_timers_count += 1
+            # ...and remove it from our pending sleep events list:
+            del self._pending_sleep_events[i]
```

Comment:
This is mutating a list being iterated over, which may cause unexpected side-effects. Deleting indexes from a list can be problematic. Suggest a different structure, dict or set...

Reply:
I converted the list into a dict, which is also a much more elegant design :ok_hand: [0ee9c57](https://github.com/Textualize/textual/pull/507/commits/0ee9c57f597b26088c076e934f7f4492c869ae6d)

Oh and while I was there I also (arbitrarily) increased the non-Windows duration of the "atomic waiting time to let asyncio do its stuff after we advanced the clock", hopefully this will make the test suite pass on your local own environment! :crossed_fingers: